### PR TITLE
Explicitly pass kubernetes.Interface to functions that need it

### PR DESCRIPTION
**Pull request by bergmannf**
## Why is this PR needed?

Supersedes #429 (it seems after the skuba rename, fork <-> origin connection is lost).

Instead of having every function retrieving the `GetAdminClientSet` explicitly this will dependency inject this into every function that needs it.

### CODE REVIEW

File: internal/pkg/skuba/deployments/ssh/cni.go [hunk 1]
```diff
@@ -33,10 +34,14 @@ func init() {
 }
 
 func cniDeploy(t *Target, data interface{}) error {
```
Comment (ereslibre) [inline internal/pkg/skuba/deployments/ssh/cni.go]: With this refactor I think we can also consider to pass the interface to all states as an argument along with the target and data.

Comment (ereslibre) [reply internal/pkg/skuba/deployments/ssh/cni.go]: Or maybe as a function/attribute on `Target`; maybe it doesn't make sense (depends on how often we use the clientset on the states themselves)

**Comment by bergmannf** [comment]
I 100% agree on not merging, I just reopened it after realizing my rebase didn't end up in the old PR.

Before I forget about it again, I wanted to make sure I have a new one open.

**Comment by jenting** [comment]
Use context and put the clientset into context is a good way.

BTW, it would be nice to align the import ordering.
// standard package

// 3rd-party package

// skuba package

**Comment by inercia** [comment]
> BTW, it would be nice to align the import ordering.
> // standard package
> 
> // 3rd-party package
> 
> // skuba package

I completely agree, I think we should standarize the order of imports in our code.

**Comment by jenting** [comment]
> > BTW, it would be nice to align the import ordering.
> > // standard package
> > // 3rd-party package
> > // skuba package
> 
> I completely agree, I think we should standarize the order of imports in our code.

I have VScode setting for auto reordering.

I think I could add CI check on this, post-GA :)

**Comment by ereslibre** [comment]
This PR has many conflicts, should it be revisited again, or closed?

**Comment by suseclee** [comment]
I am working on it with https://github.com/SUSE/avant-garde/issues/989. 
I will close it after I correct all info from this PR.

**Comment by jenting** [comment]
Close this one, another [PR](https://github.com/SUSE/skuba/pull/806) address this got merged.

**Pull request closed without merging**